### ISSUE TITLE
Incorrect error when destructuring same property twice

### ISSUE THREAD

**Issue by ajafff** [microsoft/TypeScript#35939]
**TypeScript Version:**  3.7.2

**Search Terms:** destructuring same name

**Code**

```ts
let { foo, foo: bar } = { foo: 1 };
({ foo, foo: bar } = { foo: 2 }); // error here because of duplicate property 'foo'
```

**Expected behavior:**

No error. This is totally valid code.

**Actual behavior:**

Error on the destructuring assignment:

```
An object literal cannot have multiple properties with the same name in strict mode.(1117)
Duplicate identifier 'foo'.(2300)
```

I guess this just needs a check whether the object literal is the target of an assignment.

**Playground Link:** https://www.typescriptlang.org/play/#code/DYUwLgBA3hBmD28A0dEC4ICMCGAnCAvhALzSrwYCMhA3AFAAUMCy5GO+Rpz6EATIQCUNIA
